dvc/repo/__init__.py

```diff
@@ -314,9 +314,9 @@ def collect_granular(
                 # should say that both stage name and file does not exist.
                 if file and is_valid_filename(file):
                     raise
-                raise NoOutputOrStage(target, exc.file)
+                raise NoOutputOrStageError(target, exc.file) from exc
             except StageNotFound as exc:
-                raise NoOutputOrStage(target, exc.file)
+                raise NoOutputOrStageError(target, exc.file) from exc
```

Comment:
We concat message of every chain of exceptions. So, when we get `CheckoutErrorSuggestGit`,
we will get following error message:
```
$ dvc repro random_name
ERROR: Did you mean `git checkout random_name`?: 'random_name' does not exist as an output or a stage name in 'dvc.yaml': "Stage 'random_name' not found inside 'dvc.yaml' file"
```
So, I left it out, as *context* is anyway set if they are in `try...except` block.

Reply:
Alternative thing to do is only concat messages from 2 exceptions.